code/__HELPERS/mobs.dm

```diff
@@ -186,43 +186,49 @@ GLOBAL_LIST_EMPTY(species_list)
 
 /proc/do_mob(mob/user , mob/target, time = 30, uninterruptible = 0, progress = 1, datum/callback/extra_checks = null)
 	if(!user || !target)
-		return 0
+		return FALSE
 	var/user_loc = user.loc
 
 	var/drifting = FALSE
 	if(SSmove_manager.processing_on(user, SSspacedrift))
-		drifting = 1
+		drifting = TRUE
+
+	LAZYADD(user.do_afters, target)
+	LAZYADD(target.targeted_by, user)
```

Comment:
If the user gets deleted in the middle of this, won't target.targeted_by have a deleted user in it and cause a hard-del?